public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -423,26 +431,46 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     return this._initialState;
   }
 
-  public addPanel(vizPanel: VizPanel): void {
+  public addRow(row: SceneGridRow) {
     if (!(this.state.body instanceof SceneGridLayout)) {
       throw new Error('Trying to add a panel in a layout that is not SceneGridLayout');
     }
 
     const sceneGridLayout = this.state.body;
 
-    // move all gridItems below the new one
-    for (const child of sceneGridLayout.state.children) {
```

Comment:
I realised now that grid cell sorting is done entirely in scenes and we don't need to worry about it here at all so I removed all logic for panel offsetting when something new is added

Reply:
@ivanortegaalba Heads-up! I know we talked a bit about this in the other [PR](https://github.com/grafana/grafana/pull/83145).